### PR TITLE
[FEATURE] Added option to add category

### PR DESCRIPTION
**Pull request by T1duS**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [X] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

### CODE REVIEW

File: mod_regression/controllers.py [hunk 1]
```diff
@@ -112,7 +115,20 @@ def category_edit(category_id):
...
+    """
+    form = AddCategoryForm(request.form)
+    if form.validate():
+        new_category = Category(name = form.category_name.data,description = form.category_description.data)
+        g.db.add(new_category)
+        g.db.commit()
+        flash('New Category Added')
+        form = AddCategoryForm(None)
+        return redirect(url_for('.index'))
+    else:
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py]: This else is unnecessary.

File: mod_regression/controllers.py [hunk 2]
```diff
@@ -112,7 +115,20 @@ def category_edit(category_id):
...
+    """
+    Function to add a regression test category
+    """
+    form = AddCategoryForm(request.form)
+    if form.validate():
+        new_category = Category(name = form.category_name.data,description = form.category_description.data)
+        g.db.add(new_category)
+        g.db.commit()
+        flash('New Category Added')
+        form = AddCategoryForm(None)
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py]: Why reset this if you do a redirect?

File: mod_regression/controllers.py [hunk 3]
```diff
@@ -112,7 +115,20 @@ def category_edit(category_id):
...
+@check_access_rights([Role.admin])
 def category_add():
-    # Add a regression test category
-    pass
+    """
+    Function to add a regression test category
+    """
+    form = AddCategoryForm(request.form)
+    if form.validate():
+        new_category = Category(name = form.category_name.data,description = form.category_description.data)
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py]: Spaceing isn't good here.

**Comment by T1duS** [comment]
@canihavesomecofee file edited

**Comment by canihavesomecoffee** [comment]
@T1duS Can you resolve the conflicts between your version and the current master? Aside from that, I'll need the tests to be added :+1:

**Comment by T1duS** [comment]
Resolved the conflicts. Will add unit tests tomorrow as I literally have a unit test in school tomorrow. xD :p

File: tests/test_regression/TestControllers.py [hunk 4]
```diff
@@ -77,4 +77,27 @@ def test_delete(self):
...
+    def test_add_category(self):
+        """
+        Check it will add a category
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/category_add', data=dict(name="Lost", description="And found"))
+            self.assertNotEqual(Category.query.filter(Category.name=="Lost"),None)
```
Comment (satyammittal) [inline tests/test_regression/TestControllers.py]: add .first()

**Comment by canihavesomecoffee** [comment]
Your tests aren't working as intended. There's no coverage on the new code.

File: mod_regression/controllers.py [hunk 5]
```diff
@@ -4,13 +4,17 @@
 In this module, we are trying to create, update, edit, delete and
 other various operations on regression tests.
 """
-from flask import Blueprint, g, abort, jsonify, abort, redirect, url_for
+
+from flask import Blueprint, g, abort, jsonify, abort, redirect, url_for, request, render_template, flash
```
Comment (satyammittal) [inline mod_regression/controllers.py]: Remove unnecessary imports. render_template

Comment (T1duS) [reply mod_regression/controllers.py]: done

**Pull request merged**